### PR TITLE
`ping_slot_frequency` invalid field

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR fixes part of [sentry issue](https://sentry.io/organizations/the-things-industries/issues/2298261705/events/597f313761584ce2be2361edf4ecc694/?project=2682566&query=frontendOrigin%3ATrue+is%3Aunresolved&sort=freq&statsPeriod=14d). 

#### Changes

- Change condition for showing `ping_slot_frequency` form field according to what I wrote in #tts and discussion with Adrian.
- Make sure the field value makes it into patch.

#### Testing

Manually. 
It should be possible now to enable `class B` when `ping_slot_frequency` is not already set.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Review by bafonins — APPROVED** [review]

File: pkg/webui/console/views/device-general-settings/network-server-form/index.js [hunk 1]
```diff
@@ -245,6 +245,10 @@ const NetworkServerForm = React.memo(props => {
         context: validationContext,
         stripUnknown: true,
       })
+      // Make sure `ping_slot_frequency` makes it into patch.
+      if (values.mac_settings.ping_slot_frequency) {
+        castedValues.mac_settings.ping_slot_frequency = values.mac_settings.ping_slot_frequency
+      }
```
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/index.js]: This should be solved via the validation schema in `/console/views/device-general-settings/network-server-form/validation-schema.js`

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/console/views/device-general-settings/network-server-form/index.js [hunk 2]
```diff
@@ -245,6 +245,7 @@ const NetworkServerForm = React.memo(props => {
         context: validationContext,
         stripUnknown: true,
       })
+
```
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/index.js]: Please remove.

File: pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js [hunk 3]
```diff
@@ -255,6 +255,13 @@ const validationSchema = Yup.object()
               return Yup.number().strip()
             }
 
+            if (mode === ACTIVATION_MODES.ABP) {
+              return Yup.number()
+                .min(0, Yup.passValues(sharedMessages.validateNumberGte))
+                .max(15, Yup.passValues(sharedMessages.validateNumberLte))
+                .required(sharedMessages.validateRequired)
```
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js:269]: You should additionally also conditionally set the `required` prop on the field component itself.

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js [hunk 4]
```diff
@@ -296,11 +303,13 @@ const validationSchema = Yup.object()
...
           ping_slot_frequency: Yup.lazy(frequency => {
-            if (!Boolean(frequency) || !isClassB || mode === ACTIVATION_MODES.OTAA) {
+            if (!isClassB || (mode === ACTIVATION_MODES.OTAA && !Boolean(frequency))) {
               return Yup.number().strip()
             }
 
-            return Yup.number().min(100000, Yup.passValues(sharedMessages.validateNumberGte))
+            return Yup.number()
+              .min(100000, Yup.passValues(sharedMessages.validateNumberGte))
+              .required(sharedMessages.validateRequired)
```
Comment (kschiffer) [inline pkg/webui/console/views/device-general-settings/network-server-form/validation-schema.js:324]: This is not working, I can still submit the form with an empty ping slot frequency when class B is enabled (resulting in a submit error). This should be caught on validation level already. Please check what's the issue there.

**Review by kschiffer — CHANGES_REQUESTED** [review]

**Comment by htdvisser** [comment]
Removed reviewers and labels that were automatically added before the base branch change

**Comment by ryaplots** [comment]
This PR is temporarily on hold until we come up with a good solution for this.

**Comment by kschiffer** [comment]
Let's close this until we have a plan to resolve the general issue of mac settings validation.

**Pull request closed without merging**